hack/tools/tilt-prepare/main.go

```diff
@@ -106,19 +124,73 @@ func main() {
 
 	ctx := ctrl.SetupSignalHandler()
 
+	ts, err := readTiltSettings(*tiltSettingsFileFlag)
+	if err != nil {
+		klog.Exit(fmt.Sprintf("[main] failed to read tilt settings: %v", err))
+	}
+
 	// Execute a first group of tilt prepare tasks, building all the tools required in subsequent steps/by tilt.
 	if err := tiltTools(ctx); err != nil {
 		klog.Exit(fmt.Sprintf("[main] failed to prepare tilt tools: %v", err))
 	}
 
 	// execute a second group of tilt prepare tasks, building all the resources required by tilt.
-	if err := tiltResources(ctx); err != nil {
+	if err := tiltResources(ctx, ts); err != nil {
 		klog.Exit(fmt.Sprintf("[main] failed to prepare tilt resources: %v", err))
 	}
 
 	klog.Infof("[main] completed, elapsed: %s\n", time.Since(start))
 }
 
+// readTiltSettings reads a tilt-settings.(json|yaml) file from the given path and sets debug defaults for certain
+// fields that are not present.
+func readTiltSettings(path string) (tiltSettings, error) {
+	f, err := os.Open(filepath.Clean(path))
+	if err != nil {
+		return tiltSettings{}, errors.Wrap(err, fmt.Sprintf("failed to open tilt-settings file for path: %s", path))
+	}
+	defer f.Close()
+
+	data, err := io.ReadAll(f)
+	if err != nil {
+		return tiltSettings{}, errors.Wrap(err, "failed to read tilt-settings content")
+	}
+
+	ts := tiltSettings{}
```

Comment:
Have we considered making this a pointer?

Then we can just return nil in the other return statements and also pass it to `setDebugDefaults` so the call to `setDebugDefaults` won't copy the struct. (and setDebugDefaults doesn't need a return value, cf. `PopulateDefaultsMachineDeployment`)

Reply:
Thx!